commandlineparser/src/Parser.cpp

```diff
@@ -130,3 +133,77 @@ void cmdParser::Parser::long_help(const std::vector<std::string>&keys) const
 
 	print(keys, print_long_help,"Long_Description");
 }
+
+void cmdParser::Parser::validity_checker(int argc, char *argv[])
+{
+	for (int i = 0; i < argc; i++)
+	{
+		if (argv[i][0] == '-' || argv[i][1] == '-')
+		{
+			std::stringstream ss(argv[i]);
+			std::string temp;
+			getline(ss, temp, '=');
+
+			//std::cout << temp << std::endl;
+			
+			if (command_list.find(temp) == command_list.end())
+			{
+				throw std::exception("Entered command not found in the registered command list");
+			}
+			
+			if (store_commands.find(temp) != store_commands.end())
+			{
+				store_commands.erase(temp);
+				//store_commands.insert({ temp, i });
+				store_commands.insert({ temp, i});
```

Comment:
write a comment why deleteion and insertion are done

Reply:
done.